app/jobs/geocode_location_job.rb

```diff
@@ -47,6 +51,6 @@ def enqueue_next_job
   end
 
   def next_location
-    Location.where('latitude IS NULL OR longitude IS NULL').first
+    Location.where(geocode_response: nil).first
```

Comment:
@anandaroop `geocode_response` is `nil` when the location has not been geocoded. But, once selected by the geocoder, there are only two code paths: error or result.